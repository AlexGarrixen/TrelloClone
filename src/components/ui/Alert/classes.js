const classes = {
  default: 'alert',
  // class applied if severity='success'
  severitySuccess: 'alert--success',
  // class applied if severity='info'
  severityInfo: 'alert--info',
  // class applied if severity='warning'
  severityWarning: 'alert--warning',
  // class applied if severity='error'
  severityError: 'alert--error',
};

module.exports = classes;
