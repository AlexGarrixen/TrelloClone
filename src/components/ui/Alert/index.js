import React from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';
import Icons from '../../icons';
import capitalize from '../../../utils/capitalize';
import classes from './classes';

const Alert = ({ severity = 'success', className, children, ...other }) => (
  <div
    {...other}
    className={clsx(
      classes.default,
      classes[`severity${capitalize(severity)}`],
      className
    )}
  >
    {severity === 'success' && <Icons.Check size='md' />}
    {severity === 'info' && <Icons.Info size='md' />}
    {severity === 'warning' && <Icons.Warning size='md' />}
    {severity === 'error' && <Icons.Error size='md' />}
    <div className='alert__message'>{children}</div>
  </div>
);

Alert.propTypes = {
  severity: PropTypes.oneOf(['error', 'warning', 'info', 'success']),
  children: PropTypes.node,
};

export default Alert;
