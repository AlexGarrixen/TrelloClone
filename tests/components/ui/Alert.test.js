import React from 'react';
import Alert from '../../../src/components/ui/Alert';
import { mount } from 'enzyme';

describe('<Alert />', () => {
  test('Render success', () => {
    const wrapper = mount(<Alert>Success</Alert>);
    const alertMessage = wrapper.find('.alert__message');
    const svgElement = wrapper.find('svg');

    expect(wrapper).toMatchSnapshot();
    expect(svgElement.exists()).toBe(true);
    expect(alertMessage.exists()).toBe(true);
    expect(alertMessage.text()).toBe('Success');
  });

  test('Render with prop severity:default', () => {
    const wrapper = mount(<Alert>Success</Alert>);
    const wrapperDiv = wrapper.find('div').at(0);
    const alertMessage = wrapper.find('.alert__message');

    expect(wrapperDiv.hasClass('alert alert--success')).toBe(true);
    expect(alertMessage.exists()).toBe(true);
    expect(alertMessage.text()).toBe('Success');
  });

  test('Render with prop severity:error', () => {
    const wrapper = mount(<Alert severity='error'>Error</Alert>);
    const wrapperDiv = wrapper.find('div').at(0);
    const alertMessage = wrapper.find('.alert__message');

    expect(wrapperDiv.hasClass('alert alert--error')).toBe(true);
    expect(alertMessage.exists()).toBe(true);
    expect(alertMessage.text()).toBe('Error');
  });

  test('Render with prop severity:info', () => {
    const wrapper = mount(<Alert severity='info'>Info</Alert>);
    const wrapperDiv = wrapper.find('div').at(0);
    const alertMessage = wrapper.find('.alert__message');

    expect(wrapperDiv.hasClass('alert alert--info')).toBe(true);
    expect(alertMessage.exists()).toBe(true);
    expect(alertMessage.text()).toBe('Info');
  });

  test('Render with prop severity:warning', () => {
    const wrapper = mount(<Alert severity='warning'>Warning</Alert>);
    const wrapperDiv = wrapper.find('div').at(0);
    const alertMessage = wrapper.find('.alert__message');

    expect(wrapperDiv.hasClass('alert alert--warning')).toBe(true);
    expect(alertMessage.exists()).toBe(true);
    expect(alertMessage.text()).toBe('Warning');
  });
});
