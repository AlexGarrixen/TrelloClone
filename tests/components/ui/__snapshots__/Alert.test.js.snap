// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> Render success 1`] = `
<div
  className="alert alert--success"
>
  <span>
    <svg
      aria-hidden="true"
      className="prefix__svg-inline--fa prefix__fa-check prefix__fa-w-16"
      data-icon="check"
      data-prefix="fas"
      height={30}
      viewBox="0 0 512 512"
      width={30}
    >
      <path
        d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
        fill="currentColor"
      />
    </svg>
  </span>
  <div
    className="alert__message"
  >
    Success
  </div>
</div>
`;
